openmc/deplete/microxs.py

```diff
@@ -192,8 +202,122 @@ def __init__(self, data: np.ndarray, nuclides: List[str], reactions: List[str]):
+        if isinstance(energies, str):
+            energies = GROUP_STRUCTURES[energies]
+        else:
+            # if user inputs energies check they are ascending (low to high) as
+            # some depletion codes use high energy to low energy.
+            if not np.all(np.diff(energies) > 0):
+                raise ValueError('Energy group boundaries must be in ascending order')
+
+        # check dimension consistency
+        if len(multigroup_flux) != len(energies) - 1:
+            raise ValueError('Length of flux array should be len(energies)-1')
+
+        chain_file_path = _resolve_chain_file_path(chain_file)
+        chain = Chain.from_xml(chain_file_path)
+
+        cross_sections = _find_cross_sections(model=None)
+        nuclides_with_data = _get_nuclides_with_data(cross_sections)
+
+        # If no nuclides were specified, default to all nuclides from the chain
+        if not nuclides:
+            nuclides = chain.nuclides
+            nuclides = [nuc.name for nuc in nuclides]
+
+        # Get reaction MT values. If no reactions specified, default to the
+        # reactions available in the chain file
+        if reactions is None:
+            reactions = chain.reactions
+        mts = [REACTION_MT[name] for name in reactions]
+
+        # Normalize multigroup flux
+        multigroup_flux = np.asarray(multigroup_flux)
+        multigroup_flux /= multigroup_flux.sum()
+
+        # Create 2D array for microscopic cross sections
+        microxs_arr = np.zeros((len(nuclides), len(mts)))
+
+        with TemporaryDirectory() as tmpdir:
+            # Create a material with all nuclides
+            mat_all_nucs = openmc.Material()
+            for nuc in nuclides:
+                if nuc in nuclides_with_data:
+                    mat_all_nucs.add_nuclide(nuc, 1.0)
+            mat_all_nucs.set_density("atom/b-cm", 1.0)
+
+            # Create simple model containing the above material
+            surf1 = openmc.Sphere(boundary_type="vacuum")
+            surf1_cell = openmc.Cell(fill=mat_all_nucs, region=-surf1)
+            model = openmc.Model()
+            model.geometry = openmc.Geometry([surf1_cell])
+            model.settings = openmc.Settings(
+                particles=1, batches=1, output={'summary': False})
+
+            with change_directory(tmpdir):
+                # Export model within temporary directory
+                model.export_to_model_xml()
+
+                with openmc.lib.run_in_memory(**init_kwargs):
+                    # For each nuclide and reaction, compute the flux-averaged
+                    # cross section
+                    for nuc_index, nuc in enumerate(nuclides):
+                        if nuc not in nuclides_with_data:
+                            continue
+                        lib_nuc = openmc.lib.nuclides[nuc]
+                        for mt_index, mt in enumerate(mts):
+                            xs = lib_nuc.collapse_rate(
```

Comment:
The flux is normalized in this method so that the sum of the values is one. However, now that I'm thinking about it, what we really want is a normalization such that $\sum\limits_g \int_{E_{g+1}}^{E_g} \phi(E) = 1$. Do others agree?